util/src/main/java/com/google/cloud/hadoop/util/BaseAbstractGoogleAsyncWriteChannel.java

```diff
@@ -260,6 +262,14 @@ private T waitForCompletionAndThrowIfUploadFailed() throws IOException {
       exception.addSuppressed(e);
       throw exception;
     } catch (ExecutionException e) {
+      GoogleJsonResponseException exception =
+          ApiErrorExtractor.getJsonResponseException(e.getCause());
+      if (exception != null && exception.getStatusCode() == 412) {
```

Comment:
We need to create a new method in `ApiErrorExtractor` to check this (see other methods in this class for an example)

Reply:
N/A anymore